packages/core/nano/src/core.ts

```diff
@@ -173,5 +191,9 @@ export const ops: OpContext = {
     negate: liftUnaryOp((x: any) => -x),
 };
 
-export type Formula = <O>(host: O, context: CalcValue<O>) => [Pending<unknown>[], Delayed<CalcValue<O>>];
+export type Formula = <O>(origin: O, context: CalcObj<O>) => [Pending<unknown>[], Delayed<CalcValue<O>>];
```

Comment:
I couldn't think of a good reason for having a non-object as the context for a formula, so I changed this.